app/code/Magento/CatalogGraphQl/Model/Resolver/Products/Query/Search.php

```diff
@@ -148,4 +159,27 @@ private function buildSearchCriteria(array $args, ResolveInfo $info): SearchCrit
 
         return $searchCriteria;
     }
+
+    /**
+     * Get product attribute value
+     *
+     * @param Product $product
+     * @param array $productArray
+     * @param string $field
+     *
+     * @return string|null|int|bool
+     */
+    private function getAttributeValue(Product $product, array $productArray, string $field)
+    {
+        if ($attribute = $product->getResource()->getAttribute($field)) {
```

Comment:
I wonder what performance impact will this have
Ideally we would have a collection of all attributes needed in the query, load it once and use it for any product that needs it, like we do with products where first we add to collection and we return a function instead of loading at the same time getResource()->getAttribute($field))
See this code how to do this pattern, it's the same concept: \Magento\ConfigurableProductGraphQl\Model\Resolver\Options::resolve

Also we tried to stay away from $product->getResource()
instead we would inject the resource - this way it will make the performance problem more obvious 
I can try to run this to see how much degradation it has, though I don't think we have by default such attributes that we test